vendor/diff-parsers-std-ext.patch

```diff
@@ -0,0 +1,308 @@
++    val list: ?loc:loc -> ?attrs:attrs -> pattern list -> pattern
+     val or_: ?loc:loc -> ?attrs:attrs -> pattern -> pattern -> pattern
+     val constraint_: ?loc:loc -> ?attrs:attrs -> pattern -> core_type -> pattern
+     val type_: ?loc:loc -> ?attrs:attrs -> lid -> pattern
+     val lazy_: ?loc:loc -> ?attrs:attrs -> pattern -> pattern
+     val unpack: ?loc:loc -> ?attrs:attrs -> str_opt -> pattern
+@@@@
+                 -> expression option -> expression
+     val field: ?loc:loc -> ?attrs:attrs -> expression -> lid -> expression
+     val setfield: ?loc:loc -> ?attrs:attrs -> expression -> lid -> expression
+                   -> expression
+     val array: ?loc:loc -> ?attrs:attrs -> expression list -> expression
++    val list: ?loc:loc -> ?attrs:attrs -> expression list -> expression
+     val ifthenelse: ?loc:loc -> ?attrs:attrs -> expression -> expression
+                     -> expression option -> expression
+     val sequence: ?loc:loc -> ?attrs:attrs -> expression -> expression
+                   -> expression
+     val while_: ?loc:loc -> ?attrs:attrs -> expression -> expression
+--- ocaml-4.13/ast_mapper.ml
++++ ocaml-4.13-extended/ast_mapper.ml
+@@@@
+         field ~loc ~attrs (sub.expr sub e) (map_loc sub lid)
+     | Pexp_setfield (e1, lid, e2) ->
+         setfield ~loc ~attrs (sub.expr sub e1) (map_loc sub lid)
+           (sub.expr sub e2)
+     | Pexp_array el -> array ~loc ~attrs (List.map (sub.expr sub) el)
++    | Pexp_list el -> list ~loc ~attrs (List.map (sub.expr sub) el)
+     | Pexp_ifthenelse (e1, e2, e3) ->
+         ifthenelse ~loc ~attrs (sub.expr sub e1) (sub.expr sub e2)
+           (map_opt (sub.expr sub) e3)
+     | Pexp_sequence (e1, e2) ->
+         sequence ~loc ~attrs (sub.expr sub e1) (sub.expr sub e2)
+@@@@
+              (fun (vl, p) -> List.map (map_loc sub) vl, sub.pat sub p)
+              p)
+     | Ppat_variant (l, p) -> variant ~loc ~attrs l (map_opt (sub.pat sub) p)
+     | Ppat_record (lpl, cf) ->
+         record ~loc ~attrs
+-               (List.map (map_tuple (map_loc sub) (sub.pat sub)) lpl) cf
++               (List.map (map_tuple (map_loc sub) (sub.pat sub)) lpl)
++               (match cf with
++                | Closed -> Closed
++                | Open loc -> Open (sub.location sub loc))
+     | Ppat_array pl -> array ~loc ~attrs (List.map (sub.pat sub) pl)
++    | Ppat_list pl -> list ~loc ~attrs (List.map (sub.pat sub) pl)
+     | Ppat_or (p1, p2) -> or_ ~loc ~attrs (sub.pat sub p1) (sub.pat sub p2)
+     | Ppat_constraint (p, t) ->
+         constraint_ ~loc ~attrs (sub.pat sub p) (sub.typ sub t)
+     | Ppat_type s -> type_ ~loc ~attrs (map_loc sub s)
+     | Ppat_lazy p -> lazy_ ~loc ~attrs (sub.pat sub p)
+--- ocaml-4.13/asttypes.mli
++++ ocaml-4.13-extended/asttypes.mli
+@@@@
+ 
+ type override_flag = Override | Fresh
+ 
+ type closed_flag = Closed | Open
+ 
++type closed_flag_loc = Closed | Open of Location.t
```

Comment:
Why define a new type instead of patching the one above ?

Reply:
Because this type is used in other places where it was not necessary for ocamlformat to have a location (and I wanted to minimize the diff). e.g. Ptyp_variant: the open case is `[> `A|`B ]` where there can't be a comment between `[` and `>` anyway.